File: booking_hotel_backend/src/main/java/com/group/hotel/service/ReviewService.java
package com.group.hotel.service;

import com.group.hotel.common.response.PageResponse;
import com.group.hotel.dto.request.review.ReviewCreateRequest;
import com.group.hotel.dto.request.review.ReviewReplyRequest;
import com.group.hotel.dto.request.review.ReviewSearchRequest;
import com.group.hotel.dto.response.review.ReviewResponse;
import org.springframework.data.domain.Pageable;

public interface ReviewService {
    ReviewResponse createReview(Long bookingId, ReviewCreateRequest request);
    ReviewResponse replyReview(Long reviewId, ReviewReplyRequest request);
    PageResponse<ReviewResponse> getAll(ReviewSearchRequest searchRequest, Pageable pageable);
    PageResponse<ReviewResponse> getMyReviews(ReviewSearchRequest searchRequest, Pageable pageable);
    ReviewResponse getByBookingId(Long bookingId);
    void deleteReview(Long reviewId);
}

File: booking_hotel_frontend/src/services/reviewService.js
import axiosInstance from "./axiosInstance";

const reviewService = {
    getAll: async (params) => {
        const response = await axiosInstance.get("/reviews", { params });
        return response.data;
    },
    reply: async (reviewId, data) => {
        const response = await axiosInstance.put(`/reviews/${reviewId}/reply`, data);
        return response.data;
    },
    getMyReviews: async (params) => {
        const response = await axiosInstance.get("/customer/reviews", { params });
        return response.data;
    },
    deleteReview: (reviewId) => axiosInstance.delete(`/reviews/${reviewId}`),
};

export default reviewService;
